client/src/components/saveButton.tsx

```diff
@@ -4,10 +4,19 @@ import Button from './button';
 import EditorModal from '../views/editorModal';
 import EditSvg from '../art/editSvg';
 
-export default class SaveButton extends Base {
+interface SaveButtonProps {
+    onSave: Function
+    item: {}
+}
+
+interface SaveButtonStates {
+    showEditor: boolean;
+}
+
+export default class SaveButton extends Base<SaveButtonProps, SaveButtonStates> {
     render() {
         const {onSave, item} = this.props;
-        const {showEditor} = this.state || {};
+        const {showEditor = null} = this.state || {};
```

Comment:
Just curious, what was the reason for this change?

Reply:
There is a typing error if you don't provide a default value. Because showEditor may get the value from {} when this.state is undefined. And {} doesn't have  showEditor in it. I can also write it as this.state || {showEditor: null} which is the same thing.

Reply:
Ah, yep. Makes sense.